core/src/main/java/hudson/model/Queue.java

```diff
@@ -1522,8 +1532,44 @@ public void maintain() {
     private @CheckForNull Runnable makeBuildable(final BuildableItem p) {
         if (p.task instanceof FlyweightTask) {
             if (!isBlockedByShutdown(p.task)) {
+
+                Runnable runnable = makeFlyWeightTaskBuildable(p);
+
+                if(runnable!=null){
+                    return runnable;
+                }
+
+                //this is to solve JENKINS-30084: the task has to be buildable to force the provisioning of nodes.
+                //if the execution gets here, it means the task could not be scheduled since the node
+                //the task is supposed to run on is offline or not available.
+                return new Runnable() {
```

Comment:
can this be extracted to separate method to wrap `this` as Runnable? It can reduce the number of inlined anonymous classes and increase readability